pkg/rpc/rpc.go

```diff
@@ -45,10 +45,9 @@ type RPC struct {
 	lastRequestId   uint64
 	requestsMu      sync.Mutex
 
-	errChannel        chan struct{} // never transports a value, only closed through setErr() to signal an occurred error
-	err               *Error        // only initialized via setErr(), if a rpc (Fatal/non-recoverable) error has occurred
-	errOnce           sync.Once
-	requestedShutdown bool
+	errChannel chan struct{} // never transports a value, only closed through setErr() to signal an occurred error
+	err        *Error        // only initialized via setErr(), if a rpc (Fatal/non-recoverable) error has occurred
```

Comment:
I'd rename those to something like `processResponsesErr(Ch)` as these represent whether that loop has terminated due to an error (counting orderly shutdown due to EOF as an error here).